packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -1352,14 +1365,41 @@ export class AzureCommunicationCallAdapter<AgentType extends CallAgent | TeamsCa
 
   /* @conditional-compile-remove(breakout-rooms) */
   private assignedBreakoutRoomUpdated(breakoutRoom: BreakoutRoom): void {
+    if (this.call?.id === undefined) {
+      return;
+    }
     if (breakoutRoom.state === 'closed') {
       this.returnFromBreakoutRoom();
     }
   }
 
   /* @conditional-compile-remove(breakout-rooms) */
   private breakoutRoomJoined(call: Call | TeamsCall): void {
-    this.processNewCall(call);
+    if (this.call?.id !== call.id) {
+      this.processNewCall(call);
+    }
+    // Hang up other breakout room calls in case we are joining a new breakout room while already in one
+    this.hangupOtherBreakoutRoomCalls(call);
+  }
+
+  /* @conditional-compile-remove(breakout-rooms) */
+  private hangupOtherBreakoutRoomCalls(breakoutRoomCall: Call | TeamsCall): void {
```

Comment:
rename breakoutRoomCall to -> currentBreakoutRoomCall

Reply:
nit: just pass `currentBreakoutRoomCallId` if just the id is used